movement/movement.c

```diff
@@ -294,24 +294,30 @@ void movement_request_wake() {
 }
 
 void movement_play_signal(void) {
-    bool buzzer_enabled = watch_is_buzzer_or_led_enabled();
-    if (!buzzer_enabled) {
-        watch_enable_buzzer();
-    }
-    watch_buzzer_play_note(BUZZER_NOTE_C8, 75);
-    watch_buzzer_play_note(BUZZER_NOTE_REST, 100);
-    watch_buzzer_play_note(BUZZER_NOTE_C8, 100);
-    if (!buzzer_enabled) {
-        watch_disable_buzzer();
-    }
-}
-
-void movement_play_tune(void) {
-    if (!watch_is_buzzer_or_led_enabled()) {
-        watch_enable_buzzer();
-        watch_buzzer_play_sequence(signal_tune, watch_disable_buzzer);
-    } else {
-        watch_buzzer_play_sequence(signal_tune, NULL);
+    watch_enable_buzzer();
+    watch_buzzer_play_sequence(signal_tune, watch_disable_buzzer);
+    if (movement_state.le_mode_ticks == -1) {
+	// This is somewhat of a hack. In order to play a sequence, we need to
+	// be awake. We should ideally be able to tell movement that we need to
+	// be awake for a given amount of time, but there's no good way to do
+	// this, so we block instead. This might be bad in the case that a
```

Comment:
that *is* probably the ultimate fix here: make the Movement main loop understand the the amount of awake-time necessary before going back into LE mode...